test/others/test_http_client.py

```diff
@@ -0,0 +1,73 @@
+import time
+
+import pytest
+import requests
+from requests import RequestException, HTTPError
+
+from haystack.errors import OpenAIUnauthorizedError, OpenAIRateLimitError
+from haystack.utils.http_client import HTTPClient
+
+
+@pytest.mark.unit
```

Comment:
I admit I have mixed feelings about these markers (I'm not sure these are really unit tests, because they're using a local server instead of mocking it), but let's leave them like this for now. @silvanocerza fyi

Reply:
If a test executes an HTTP request it's not a unit test for me, even if it's local.
I plan to block the use of `requests` in unit tests with #4433.